### PR TITLE
UI for Updating Next Installment Amount

### PR DESCRIPTION
**Pull request by lparrott**
- [W-11556798](https://gus.lightning.force.com/a07EE000013O0YPYA0)
- Test steps can be found in the [Design Doc](https://salesforce.quip.com/5g9hA7TLw3yS#temp:C:aVE0204ea5524d74a7fa2581cbc9)

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: force-app/main/default/lwc/rd2UpdateNextPayment/rd2UpdateNextPayment.html [hunk 1]
```diff
@@ -0,0 +1,103 @@
+<template>
+
+    <header class="slds-modal__header slds-clearfix slds-align_absolute-center" data-qa-locator="heading">
```
[the remark this answers is not in the document]
Comment (lparrott) [reply force-app/main/default/lwc/rd2UpdateNextPayment/rd2UpdateNextPayment.html:3]: I hadn't heard of this, pushing an update shortly!

**Review by andrewyu-salesforce — APPROVED** [review]
looks good

File: force-app/main/default/lwc/rd2UpdateNextPayment/rd2UpdateNextPayment.js [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+        this.dispatchEvent( new CloseActionScreenEvent() );
+    }
+
+    /***
+    * @description Records the latest Next Payment Amount value
+    * and checks if the [Save] button should be enabled.
+    */
+    handleNextPaymentAmountChange(event) {
+        this.nextPaymentAmount = event.detail.value;
```
Comment (voduyemi) [inline force-app/main/default/lwc/rd2UpdateNextPayment/rd2UpdateNextPayment.js:257]: Also just a non-blocking comment here, this function could be broken down so its easy for us to figure out what it does.

**Review by voduyemi — APPROVED** [review]
Looks good! I left some comments around breaking down some JS functions

File: force-app/main/default/classes/RD2_RecurringDonation.cls [hunk 3]
```diff
@@ -160,11 +160,21 @@ public inherited sharing class RD2_RecurringDonation {
...
     /**
-    * @description Returns True if the Next Payment Date has changed on an Elevate connected RD
-    * @return Boolean True if the Next Payment Date has changed on an Elevate connected RD
+    * @description Returns True if the Next Payment Date has changed and the value has not
+    * @return Boolean True if the Next Payment Date has changed and the value has not
     */
     public Boolean needsNextPaymentScheduleCheck() {
-        return changedFields.contains(npe03__Recurring_Donation__c.npe03__Next_Payment_Date__c);
+        return changedFields.contains(npe03__Recurring_Donation__c.npe03__Next_Payment_Date__c)
+            && !currentOrNextYearValueChanged();
```
Comment (lparrott) [inline force-app/main/default/classes/RD2_RecurringDonation.cls:168]: @andrewyu-salesforce This is what I came up with to make sure we **don't** clear the Amount when the Pause schedule changes. I don't love it, but I think it should work?

**Pull request merged**